lib/bolt/plugin/aws.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+require 'json'
+
+module Bolt
+  class Plugin
+    class Aws
+      class EC2
+        attr_writer :client
+
+        def initialize(config)
+          require 'aws-sdk-ec2'
+          @config = config
+          @logger = Logging.logger[self]
+        end
+
+        def name
+          'aws::ec2'
+        end
+
+        def hooks
+          %w[inventory_targets]
+        end
+
+        def config_client(opts)
+          options = {}
+
+          if opts.key?('region')
+            options[:region] = opts['region']
+          end
+          if opts.key?('profile')
+            options[:profile] = opts['profile']
+          end
+          if @config.aws && @config.aws.key?('credentials')
```

Comment:
I would throw an error here in an else statement stating that you need to configure credentials to connect to aws.

Reply:
If credentials aren't provided in the config file, the aws sdk searches in `ENV['ACCESS_KEY_ID']` and `ENV['SECRET_ACCESS_KEY']`, followed by the default credentials location `~/.aws/credentials`. If no credentials are found, then the aws sdk throws a descriptive error that credentials are missing.

Would we want to check these default locations and throw our own error if none of them hold credentials, or require the user to specify credentials in the config? My original thought was the config file would allow for an alternate to the default location for the credentials.

Reply:
In that case I'd catch AWS error and return a `Bolt::ValidationError` stating that you need to configure credentials.

Reply:
It looks like the AWS error is caught and a `Bolt::PluginError` is raised at `bolt/inventory/group2.rb:213`.

```
213:     raise Bolt::Plugin::PluginError.new(e.message, plugin, "inventory_targets in #{@name}")
```

Reply:
Also! I think you can shorten this to just be `@config&.aws.key?('credentials')`